internal/pkg/print/print.go

```diff
@@ -0,0 +1,78 @@
+package print
+
+import (
+	"io"
+	"log/slog"
+	"os"
+
+	"github.com/spf13/cobra"
+)
+
+type VerbosityLevel string
+
+const (
+	DebugVerbosity   VerbosityLevel = "debug"
+	InfoVerbosity    VerbosityLevel = "info"
+	WarningVerbosity VerbosityLevel = "warning"
+	ErrorVerbosity   VerbosityLevel = "error"
+)
+
+type Printer struct {
+	Cmd       *cobra.Command
+	Verbosity VerbosityLevel
+}
+
+// Creates a new printer, including setting up the default logger.
+func NewPrinter() Printer {
+	logger := slog.New(slog.NewTextHandler(os.Stderr, &slog.HandlerOptions{AddSource: true, Level: slog.LevelDebug}))
+	slog.SetDefault(logger)
+
+	return Printer{}
+}
+
+// Print an output using Printf to the defined output (falling back to Stderr if not set).
+func (p *Printer) Outputf(message string, args ...any) {
+	p.Cmd.Printf(message, args...)
+}
+
+// Print an output using Println to the defined output (falling back to Stderr if not set).
+func (p *Printer) Outputln(message string) {
+	p.Cmd.Println(message)
+}
+
+// Print a Debug level log through the "slog" package.
+// If the verbosity level is not Debug, it does nothing
+func (p *Printer) Debug(message string, args ...any) {
```

Comment:
The function signature is missing the log level argument, which should be used to select which `slog` method to call. The structured debug logs can have different log levels. That's also why I commented above to rename `Verbosity` from the const

```
func (p *Printer) Debug(message string, logLevel Level, args ...any) {
	if p.Verbosity != Debug {
		return
	}
        switch logLevel {
        case Debug
	    slog.Debug(message, args...)
	case Info
	    slog.Info(message, args...)
       // and so on .....
}
```